### PR TITLE
Fix incorrect backoff-wait logic.

### PR DESCRIPTION
**Pull request by hashi-derek**
This change fixes a situation where every call to `syncViaBlockingQuery` gets exponentially slower, until reaching a cap. The change now modifies the logic to exponentially backoff and only sleep in the event of a failure. The wait time also is reset back to normal whenever a successful call is made.

### CODE REVIEW

**Comment by hashi-derek** [comment]
Not sure if I should be adding a changelog entry for beta features that aren't technically fully-released yet.

File: agent/grpc-external/services/peerstream/subscription_blocking.go [hunk 1]
```diff
@@ -93,30 +93,27 @@ func (m *subscriptionManager) syncViaBlockingQuery(
...
-
 			// Block for any changes to the state store.
 			ws.WatchCtx(ctx)
 		}
-
-		err := waiter.Wait(ctx)
-		if errors.Is(err, context.Canceled) || errors.Is(err, context.DeadlineExceeded) {
-			return
-		} else if err != nil {
-			logger.Error("failed to wait before re-trying sync", "error", err)
```
Comment (huikang) [inline agent/grpc-external/services/peerstream/subscription_blocking.go:119]: Are we missing the case to emit this logger error?

File: agent/grpc-external/services/peerstream/subscription_blocking.go [hunk 2]
```diff
@@ -93,30 +93,27 @@ func (m *subscriptionManager) syncViaBlockingQuery(
 
 	store := m.getStore()
 
-	for {
+	for ctx.Err() == nil {
```
Comment (huikang) [inline agent/grpc-external/services/peerstream/subscription_blocking.go:96]: Could you elaborate why `ctx.Err() == nil` is in the for condition? In the original version, the error msg will be caught by 
```go
} else if err != nil {
			logger.Error("failed to wait before re-trying sync", "error", err)
```
Now we seem to miss handling the error.

File: agent/grpc-external/services/peerstream/subscription_blocking.go (hunk 1 above)
Comment (hashi-derek) [reply agent/grpc-external/services/peerstream/subscription_blocking.go:119]: The `waiter.Wait` command can only error whenever the context is cancelled, according to the documentation on the method, so the old `else-if err != nil` should be encountered ever. The new implementation will simply stop and break the loop quietly, just like the old one.

Comment (freddygv) [reply agent/grpc-external/services/peerstream/subscription_blocking.go:119]: Good catch

**Review by freddygv — APPROVED** [review]
This looks good to me, but maybe wait for @huikang to confirm that it resolves the issue before merging

**Review by huikang — APPROVED** [review]
The PR fixes the performance issue, reducing the exporting latency to ~200ms. 
However, at certain point, exporting to certain peering cluster will be blocked. But I think this is a different story and I will document elsewhere. That being said, this PR can be merged. Thanks for quick fix, @hashi-derek

**Pull request merged**